packages/repository/README.md

```diff
@@ -21,47 +21,137 @@ $ npm install --save @loopback/repository
+3.x `loopback-datasource-juggler` and connectors. The following steps illustrate
+how to define repositories and use them with controllers.
+
+### Defining a legacy datasource and a model
+
 The repository module provides APIs to define LoopBack 3.x data sources and
 models. For example,
 
 ```ts
-import {
-  DataSourceConstructor,
-  juggler,
-  Entity,
-  model,
-  ModelDefinition
-} from '@loopback/repository';
+// src/datasources/db.datasource.ts
+import {juggler, DataSourceConstructor} from '@loopback/repository';
 
-export const ds: juggler.DataSource = new DataSourceConstructor({
+export const db: juggler.DataSource = new DataSourceConstructor({
   name: 'db',
   connector: 'memory',
 });
+```
+
+```ts
+// src/models/note.model.ts
+import {model, Entity, property} from '@loopback/repository';
 
 @model()
 export class Note extends Entity {
-  static definition = new ModelDefinition({
-    name: 'note',
-    properties: {
-      id: {name: 'id', type: 'number', id: true},
-      title: 'string',
-      content: 'string'
-    }
-  })
-};
+  @property({id: true})
+  id: string;
+  @property() title: string;
+  @property() content: string;
+}
+```
+
+**NOTE**: There is no declarative support for data source and model yet in
+LoopBack 4. These constructs need to be created programmatically as
+illustrated above.
+
+### Defining a repository
+
+A repository can be created by extending `DefaultCrudRepository` and using
+dependency injection to resolve the datasource.
+
+```ts
+// src/repositories/note.repository.ts
+import {DefaultCrudRepository, DataSourceType} from '@loopback/repository';
+import {Note} from '../models';
+import {inject} from '@loopback/core';
+
+export class NoteRepository extends DefaultCrudRepository<
+  Note,
+  typeof Note.prototype.id
+> {
+  constructor(@inject('datasources.db') protected dataSource: DataSourceType) {
+    super(Note, dataSource);
+  }
+}
 ```
-A repository can be created directly using `DefaultCrudRepository`.
+
+### Defining a controller
+
+Controllers serve as handlers for API requests. We declare controllers as
+classes with optional dependency injection by decorating constructor parameters
+or properties.
 
 ```ts
-import {DefaultCrudRepository} from '@loopback/repository';
-// also import Note and ds from wherever you defined them
+// src/controllers/note.controller.ts
+import {repository} from '@loopback/repository';
+import {NoteRepository} from '../repositories';
+import {Note} from '../models';
+import {post, requestBody, get, param} from '@loopback/openapi-v3';
+
+export class NoteController {
+  constructor(
+    // Use constructor dependency injection to set up the repository
+    @repository(NoteRepository.name) public noteRepo: NoteRepository,
+  ) {}
+
+  // Create a new note
+  @post('/note')
+  create(@requestBody() data: Note) {
+    return this.noteRepo.create(data);
+  }
+
+  // Find notes by title
+  @get('/note/{title}')
+  findByTitle(@param.query.string('title') title: string) {
```

Comment:
Since your path contains `{title}` in it, you should not use `param.query`. I think `param.path` is what you need.